alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectionInfoCacheTest.java

```diff
@@ -237,209 +184,22 @@ public void testGetConnectionInfo_scheduledNextOperationImmediately_onCertificat
             connectionInfoRepo,
             instanceName,
             keyPair,
-            new RefreshCalculator(),
-            spyRateLimiter);
-
-    executor.runNextPendingCommand(); // Simulate completion of background thread.
-    RuntimeException runtimeException =
-        assertThrows(RuntimeException.class, connectionInfoCache::getConnectionInfo);
-
-    // The RuntimeException wraps an ExecutionException which wraps the original exception.
-    assertThat(runtimeException.getCause().getCause()).isInstanceOf(CertificateException.class);
-
-    executor.tick(1, TimeUnit.SECONDS); // Advance time just a little
-    executor.runUntilIdle(); // Simulate completion of background thread.
-
-    ConnectionInfo connectionInfo = connectionInfoCache.getConnectionInfo();
-
-    assertThat(
-            connectionInfo
-                .getClientCertificate()
-                .getNotAfter()
-                .toInstant()
-                .truncatedTo(ChronoUnit.SECONDS))
-        .isEqualTo(ONE_HOUR_FROM_NOW.truncatedTo(ChronoUnit.SECONDS));
-  }
-
-  @Test
-  public void testGetConnectionInfo_isRateLimited() {
```

Comment:
The rate limiter is internal to the Refresher and can't be tested here. I added `RefresherTest.testRateLimiterInUse()` to test this.